### DISCUSSION TITLE
DuckDB Version Mismatch (Trying to read a database file with version number XX)

### DISCUSSION THREAD

**Discussion by shreeve** [tconbeer/harlequin#257] [Q&A, Troubleshooting, Support]
The `harlequin` CLI is amazing!

The newest version of DuckDB v0.9.0 is not quite yet supported though:

```
╭─ DuckDB couldn't connect to your database. ──────────────────────────────────╮
│ IO Error: Trying to read a database file with version number 64, but we can  │
│ only read version 51.                                                        │
│ The database file was created with an newer version of DuckDB.               │
│                                                                              │
│ The storage of DuckDB is not yet stable; newer versions of DuckDB cannot     │
│ read old database files and vice versa.                                      │
│ The storage will be stabilized when version 1.0 releases.                    │
│                                                                              │
│ For now, we recommend that you load the database file in a supported version │
│ of DuckDB, and use the EXPORT DATABASE command followed by IMPORT DATABASE   │
│ on the current version of DuckDB.                                            │
│                                                                              │
│ See the storage page for more information:                                   │
│ https://duckdb.org/internals/storage                                         │
╰───────────────────────────────────────────────────────────────── Try again? ─╯
```

**Comment by tconbeer** [comment]
Hi,
Thanks for your support and feedback. 

Harlequin works fine with 0.9.0 if you create  database with 0.9.0. The error shown here is the same as you get from the DuckDB CLI or the Python API if you attempt to open a database file created by a different version of DuckDB. 

You can pin the version of DuckDB that Harlequin uses by adding a pinned, direct dependency on DuckDB to your project. If you used pipx to install Harlequin, you can "inject" that dependency:

```
pipx inject harlequin duckdb==0.8.1
```

**Comment by tconbeer** [comment]
Sorry, just re-read your error message and my advice is the same, but in this case you need to upgrade (not downgrade) the version of DuckDB installed by Harlequin. You could do this by pinning the new version or just updating/upgrading/reinstalling Harlequin. (A new version will be out tomorrow)

**Comment by shreeve** [comment]
Fantastic! Thank you for an absolutely phenomenal tool @tconbeer !!!
